### ISSUE TITLE
BGFX stops working when using artwork

### ISSUE THREAD

**Issue by SaucySarah** [Robbbert/store1#8]
I use all default settings and only change 'Video Mode' to BGFX and 'BGFX screen chains' to crt-geom-deluxe. This works fine. If I enable artwork then BGFX isn't used and goes to the default screen mode. I had to delete the mame.ini to get it to work again. (I'm using MAME 0.202).

I tried this on regular MAME and it works fine. Do you need the verbose log ?

**Comment by Robbbert** [comment]
Verbose log doesn't say anything useful unfortunately. I did not know what to expect so tried a few things in MAME, with the error messages being quite illogical, but somehow it ran. After confirming my installation of ARCADE64 didn't work the same, I noticed the path "artwork" was missing from artpath in ini\mame.ini. Once added, it worked.

**Comment by SaucySarah** [comment]
I think i've discovered what's happening based on what you just said. In the 'Directories' option in ArcadeFX, if you select 'Artworks' from the drop down menu and then select 'Browse' and choose a different directory for the artwork then BGFX isn't used. It also needs the default 'artwork' directory in the list.

So when using BGFX, maybe it should be hardcoded to always look in the 'Arcade64/artwork' directory, because that's where the 'bgfx' directory is. If you change the 'artwork' directory then it isn't used.

Or when selecting directories for artwork, make it not possible to change the default artwork directory, you will have to add at least a second directory and not be able to replace the default.

This is why it probably works fine in regular MAME because you can't change the default where the bgfx directory is, you just add other directories for MAME to search.

**Comment by SaucySarah** [comment]
But it should always look in the default artwork directory regardless of what other directories you have set because it contains important files that are needed. Regular MAME always checks the default artwork directory as well as any others because you can't delete the default. But you probably won't read this as you've made your mind up.

**Comment by Robbbert** [comment]
> Regular MAME always checks the default artwork directory as well as any others because you can't delete the default.

Not true, you **can** delete the default. I tried it.

> But you probably won't read this as you've made your mind up.

As you can see, I did read it. But I have made up my mind. It's a user error.

**Comment by SaucySarah** [comment]
Yes you can **delete** the default in regular MAME but you can't replace it by accident when selecting a new directory (like it does in Arcade64), it just adds a new directory, very user friendly.
